analysis/Proposed_indicators.md

```diff
@@ -1,6 +1,83 @@
+
+```python
+# Local neighbourhood analysis
+def analyse_local_nh(G, orig_point, buffer=50, length = 1600, intersection_tolerance = 15):
+    # locate closest node on network to 
+    orig_node = ox.get_nearest_node(G, orig_point, return_dist=True)
+    subgraph = nx.ego_graph(G, orig_node[0], radius=length, distance='length')
+    # We project the subgraph to UTM so we can use metres as a unit when buffering
+    subgraph = ox.project_graph(subgraph)
+    su = ox.graph_to_gdfs(subgraph, nodes=False, edges=True, fill_edge_geometry=True)
+    bu = su.geometry.buffer(buffer)
+    uu = bu.geometry.unary_union
+    area_sqm = uu.area
+    area_sqkm = area_sqm*1e-06
+    stats = ox.basic_stats(subgraph, area=area_sqm, clean_intersects=True, tolerance=intersection_tolerance,
+                circuity_dist='euclidean')
+    return({'graph': subgraph,
+             'network_gdfs':su,
+             'buffer':uu,
+             'area_sqkm':area_sqkm,
+             'stats':stats,
+             'origin_node_snap_dist': orig_node[1]})
+     
+# Define some point locations over which we'll iterate
+point_locations = [(-37.808156, 144.964942),(-37.764727, 144.964057)]
+# initialise an 
+nh_estimates = []
+
+for point in point_locations:
+  start = time.time()
+  task = "Calculate local neighbourhood measures for points"
+  nh_estimates.append(analyse_local_nh(G, point))
+  log('Completed task "{}" in {:,.2f} seconds'.format(task,time.time() - start)) 
+  
+# Get some stats
+## area in sqkm
+[x['area_sqkm'] for x in nh_estimates]
+
+## intersection density per sqkm
+[x['stats']['intersection_density_km'] for x in nh_estimates]
+
+## clean intersection density per sqkm
+[x['stats']['clean_intersection_density_km'] for x in nh_estimates]
+```
+So, that makes use of the osmnx stats functionality for calculating street connectivity.  
+
+The use of the function might be overkill; assuming we have a dataframe of sample points, if you coded it right there might even be a vectorised solution to carry out the operations, like extracting the 50m buffered unary union of 1600m network linestrings for a sample point as a geometry field and then use intersection analysis for each sample point's network buffer polygon with other pre-prepared resources:
+
+- use population raster (eg worldpop, or other) to derive a scaled-population density measure to serve as a proxy measure for dwelling density (e.g. take sum of population cells intersecting sausage buffer and divide by some indicative number for population per dwelling (e.g. 4) then standardise that result by dividing by the area of the sausage buffer in Hectares  --- so a result a-bit like dwellings per hectare (although technically, it isn't that.... i'm not saying this is valid, but it could be done and might serve for this purpose)
+
+- count clean network intersections generated by osmnx which intersect a sample point's sausage buffer and divide by area in km2 for street connectivity per square kilometre
```

Comment:
A clarifying question, is the `area in square-km` means local neighbourhood area (sausage buffer neighbourhood) rather than the study region area?

Reply:
This is correct for what I was describing, an approach for sampling local area street connectivity: so, take a point, trace out 1600m on the walkable network to create a "local walkable network" (ie. local for that sample point), then buffer by 50 metres.  The area of the buffered local walkable network (or 'sausage buffer'; but i find the term a bit gross!) is taken in square kilometres.  That is used as a standardising factor when considering the clean intersections which intersect that local area: ie. intersections per square kilometre, with respect to that sample points local walkable neighbourhood.  The idea then is, we do this for all sample points and then take some kind of small area averages.  We could then take a global average, or think about taking a population weighted average which is probably more meaningful.